Automattic/jetpack
# My Jetpack: Allow  rendering for non-admins

## Proposed changes:

This PR refines user permission handling across various aspects of My Jetpack, ranging from endpoint permissions to UI rendering for different user roles.

The minimum requirement is to be a contributor (able to write but not publish).

**Note: these links don't work. They should be addressed in a follow-up PR**

![image](https://github.com/Automattic/jetpack/assets/746152/efdb9098-c81a-4fe9-99d5-008fd56c4078)

<table>
<tr>
 <td><img alt="image" src="https://github.com/Automattic/jetpack/assets/746152/a36814bc-8d0e-4e0a-a8fc-f3832e50a525">
 <td> <img alt="image" src="https://github.com/Automattic/jetpack/assets/746152/1069b762-16c5-4b30-970a-fffc0893bfb1">
<tr>
 <td>Admin
 <td>Non-admin
</table>

**Commits' details**

* Allows **My Jetpack** menu item rendering for non-admins.
* Update permission callback on purchases endpoint to edit_posts.
* Make it so we don't render the plans section footer with manage and purchase links if the user is not an admin.
* Forwards a useIsAdmin variable to the frontend.

links to more information are shown and these links do not work. Will be tackled in another PR
### Other information:


## Jetpack product discussion

* p1691616562377759/1691614027.240459-slack-C01U2KGS2PQ
* p1HpG7-o03-p2#comment-65305

## Does this pull request change what data or activity we track or use?
Doesn't add new events, but starts tracking activity on the My Jetpack page for authors, editors and contributors

## Testing instructions:

* Connect the site as an admin
* Create a user with role Contributor
* Run this branch, and visit wp-admin
* Confirm the Jetpack -> My Jetpack menu item is shown
* COnfirm that visiting it works
* Confirm the plans section footer where links for managing or purchasing usually are, don't render for this user
* Confirm action buttons are not shown
* Confirm links to more information are shown and these links do not work. Will be tackled in another PR
